v4/Models/Survey.php

```diff
@@ -0,0 +1,427 @@
+<?php
+
+namespace v4\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Validation\Rule;
+use Ushahidi\App\Repository\FormRepository;
+use Ushahidi\App\Validator\LegacyValidator;
+use Ushahidi\Core\Entity\Permission;
+use Ushahidi\Core\Tool\Permissions\InteractsWithFormPermissions;
+
+class Survey extends Model
+{
+    use InteractsWithFormPermissions;
+
+    /**
+     * Add eloquent style timestamps
+     *
+     * @var boolean
+     */
+    public $timestamps = false;
+
+    /**
+     * Specify the table to load with Survey
+     *
+     * @var string
+     */
+    protected $table = 'forms';
+
+    /**
+     * Add relations to eager load
+     *
+     * @var string[]
+     */
+    protected $with = ['tasks'];
+
+    /**
+     * The attributes that should be hidden for serialization.
+     *
+     * @var  array
+     * @note this should be changed so that we either use the fractal transformer
+     * OR a policy authorizer which is a more or less accepted method to do it
+     * (which uses the same $hidden type thing but it's much nicer obviously)
+     */
+    protected $hidden = [
+        'description',
+        'icon',
+    ];
+
+    /**
+     * The attributes that should be mutated to dates.
+     *
+     * @var array
+     */
+    protected $dates = [
+        'created',
+        'updated',
+    ];
+
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'parent_id',
+        'name',
+        'description',
+        'type',
+        'disabled',
+        'require_approval',
+        'everyone_can_create',
+        'color',
+        'hide_author',
+        'hide_time',
+        'hide_location',
+        'targeted_survey',
+        'base_language',
+    ];
+
+    /**
+     * The accessors to append to the model's array form.
+     *
+     * @var array
+     */
+    protected $appends = ['can_create'];
+
+    /**
+     * The model's default values for attributes.
+     *
+     * @var array
+     */
+    protected $attributes = [
+        'type'                => 'report',
+        'require_approval'    => true,
+        'everyone_can_create' => true,
+        'hide_author'         => false,
+        'hide_time'           => false,
+        'disabled'            => false,
+        'hide_location'       => false,
+        'targeted_survey'     => false,
+    ];
+
+    /**
+     * The attributes that should be cast.
+     *
+     * @var array
+     */
+    protected $casts = [
+        'everyone_can_create' => 'boolean',
+        'hide_author'         => 'boolean',
+        'require_approval'    => 'boolean',
+        'disabled'            => 'boolean',
+    ];
+
+
+    /**
+     * Get the error messages for the defined validation rules.
+     *
+     * @return array
+     */
+    public static function validationMessages()
+    {
+        return [
```

Comment:
tempted to go for a static in a different class that we call from here instead (to keep it easy to follow). Will check if there's a laravel way I'm missing :thinking: